app/views/NewServerView/index.tsx

```diff
@@ -171,202 +79,100 @@ class NewServerView extends React.Component<INewServerViewProps, INewServerViewS
-				setBasicAuth(credentials);
-			}
-		} catch {
-			// do nothing
-		}
-	};
+	useEffect(() => {
+		EventEmitter.addEventListener('NewServer', handleNewServerEvent);
+		const backHandler = BackHandler.addEventListener('hardwareBackPress', handleBackPress);
 
-	chooseCertificate = async () => {
-		try {
-			const certificate = await SSLPinning?.pickCertificate();
-			this.setState({ certificate });
-		} catch {
-			// Do nothing
-		}
-	};
+		let keyboardShowListener: ReturnType<typeof Keyboard.addListener> | null = null;
+		let keyboardHideListener: ReturnType<typeof Keyboard.addListener> | null = null;
 
-	completeUrl = (url: string) => {
-		const parsedUrl = parse(url, true);
-		if (parsedUrl.auth.length) {
-			url = parsedUrl.origin;
+		if (isAndroid) {
+			keyboardShowListener = Keyboard.addListener('keyboardDidShow', () => setShowBottomInfo(false));
+			keyboardHideListener = Keyboard.addListener('keyboardDidHide', () => setShowBottomInfo(true));
 		}
 
-		url = url && url.replace(/\s/g, '');
-
-		if (/^(\w|[0-9-_]){3,}$/.test(url) && /^(htt(ps?)?)|(loca((l)?|(lh)?|(lho)?|(lhos)?|(lhost:?\d*)?)$)/.test(url) === false) {
-			url = `${url}.rocket.chat`;
-		}
+		return () => {
+			EventEmitter.removeListener('NewServer', handleNewServerEvent);
+			backHandler.remove();
 
-		if (/^(https?:\/\/)?(((\w|[0-9-_])+(\.(\w|[0-9-_])+)+)|localhost)(:\d+)?$/.test(url)) {
-			if (/^localhost(:\d+)?/.test(url)) {
-				url = `http://${url}`;
-			} else if (/^https?:\/\//.test(url) === false) {
-				url = `https://${url}`;
+			if (isAndroid) {
+				keyboardShowListener?.remove();
+				keyboardHideListener?.remove();
 			}
-		}
-		return serializeAsciiUrl(url.replace(/\/+$/, '').replace(/\\/g, '/'));
-	};
-
-	uriToPath = (uri: string) => uri.replace('file://', '');
 
-	handleRemove = () => {
-		showConfirmationAlert({
-			message: I18n.t('You_will_unset_a_certificate_for_this_server'),
-			confirmationText: I18n.t('Remove'),
-			onPress: () => this.setState({ certificate: null }) // We not need delete file from DocumentPicker because it is a temp file
-		});
-	};
-
-	deleteServerHistory = async (item: TServerHistoryModel) => {
-		const db = database.servers;
-		try {
-			await db.write(async () => {
-				await item.destroyPermanently();
-			});
-			this.setState((prevstate: INewServerViewState) => ({
-				serversHistory: prevstate.serversHistory.filter(server => server.id !== item.id)
-			}));
-		} catch {
-			// Nothing
-		}
-	};
-
-	renderCertificatePicker = () => {
-		const { certificate, showBottomInfo } = this.state;
-		const { theme, connecting } = this.props;
-
-		if (!showBottomInfo) return <></>;
-
-		return (
-			<>
-				<Text style={[styles.buttonPrompt, { color: themes[theme].fontSecondaryInfo }]}>
-					{certificate ? I18n.t('Your_certificate') : I18n.t('Do_you_have_a_certificate')}
+			if (previousServer) {
+				dispatch(serverFinishAdd());
+			}
+		};
+	}, []);
+
+	useEffect(() => {
+		setHeader();
+	}, [connecting]);
```

Comment:
Not sure, but I think...

```suggestion
	}, [connecting, previousServer]);
```